src/traversal.js

```diff
@@ -0,0 +1,34 @@
+import { base58btc } from 'multiformats/bases/base58'
+
+/**
+ * @typedef {import('./cid.js').CID} CID
+ */
+
+/**
+ * @template T
+ * @typedef {import('./block.js').Block<T>} Block
+ */
+
+/**
+ * @template T
+ * @param {Object} options
+ * @param {CID} options.cid
+ * @param {(cid: CID) => Promise<Block<T>>} options.load
```

Comment:
@patrickwoodhead let's go with the `null` == skip feature for this.

```suggestion
 * @param {(cid: CID) => Promise<Block<T>|null>} options.load
```

Then you can just do a `return` if `block === null` on the `load()` call.

If you could also add some very basic docs onto the README for this that would be helpful too. As you can see the current README is fairly minimal so no need to go overboard unless you're enthusiastic (it's a TODO #47).

Since I've force pushed a rebase and some additional commits to this branch, you'll have to get your local copy updated. `git fetch origin` and `git rebase origin feat/walk` should do the trick, then you can just push new updates.

When we merge, we'll squash it down to a single commit with a `feat:` prefix so this will be a semver-minor release.

Reply:
I have done the above. Let me know if there is anything still outstanding.